src/com/google/cose/Ec2SigningKey.java

```diff
@@ -46,31 +46,33 @@ public Ec2SigningKey(DataItem cborKey) throws CoseException, CborException {
     super(cborKey);
 
     keyPair = getKeyPairFromCbor();
-    if ((operations == null)
-        || (operations.contains(Headers.KEY_OPERATIONS_VERIFY)
-        && operations.contains(Headers.KEY_OPERATIONS_SIGN))) {
-      return;
+
+    if ((operations != null)
+        && !operations.contains(Headers.KEY_OPERATIONS_VERIFY)
+        && !operations.contains(Headers.KEY_OPERATIONS_SIGN)) {
+      throw new CoseException("Signing key requires either sign or verify operation.");
     }
-    throw new CoseException("Signing key requires sign and verify operations.");
   }
 
   static class Builder {
     private String keyId;
     private Algorithm algorithm;
-    private final List<Integer> operations;
+    private final Set<Integer> operations;
     private byte[] baseIv;
     private Integer curve = null;
     private byte[] xCor;
     private byte[] yCor;
-    private byte[] dCor;
+    private byte[] dParameter;
 
     Builder() {
-      operations = new ArrayList<>();
+      operations = new HashSet<>();
     }
 
     public Ec2SigningKey build() throws CoseException, CborException {
-      if ((curve == null) || (dCor == null && (xCor == null || yCor == null))) {
-        throw new CoseException("Need curve and key material information.");
+      if (curve == null) {
+        throw new CoseException("Need curve information.");
+      } else if (dParameter == null && (xCor == null || yCor == null)) {
```

Comment:
super nitpick: `else` here is kind of out of place. None of the other conditionals are `else if`, they're all just `if`.